tests/code_update.py

```diff
@@ -43,14 +43,23 @@ def run(args):
 
         LOG.info(f"Adding a node with unsupported code id {new_code_id}")
         try:
-            network.create_and_trust_node(args.patched_file_name, "localhost", args)
+            network.create_and_add_pending_node(
+                args.patched_file_name, "localhost", args, timeout=3
```

Comment:
So how does 3 work here? We used to wait the default amount before, does 3 work because we/OE check mrenclave first and fail fast if it matches, whereas a full quote verification can take longer?

Reply:
The node-to-node join request has always failed quickly (I believe much quicker than a full quote verification). But this code doesn't look for that error (in the sending node's log) until the timeout has elapsed. So in the old case we waited for ~60s, while the node repeatedly polled the network and was constantly told "bad code ID", then finally checked the logs. Now we leave it for only a few seconds before failing and checking the logs.

Reply:
I agree. This will work for the wrong reason I think if the quote verification takes too long. We first verify the quote and then check MRENCLAVE, mostly because `oe_verify_report()` returns the parsed report.

Reply:
Why do we even retry if the code id doesn't check out? Should we just terminate, with an error code, and then we can check for that rather than the log?

Reply:
@jumaffre I don't think it will work for the wrong reasons because we check the log for explicitly this, but if this does verification first, then I don't see why shrinking the timeout is justified.

Reply:
Good point. I think we should just `join_timer.reset();` in `initiate_join()` if an error occurred?

Reply:
I think there's a few angles on this, written up as #1014.

If this fails because of a slow quote retrieval, it will fail with a different error type (a `TimeoutError` which will now propagate past this code and be seen elsewhere). I'd like to leave this timeout reduced - if we see slow quote retrieval failures we can either prioritise #1014 or increase this timeout. But waiting a long time _after_ failure to detect the failure, multiple times per CI run, is hopefully something we can avoid.

Reply:
@eddyashton Do you mean slow quote verification on the primary side? Since https://github.com/microsoft/CCF/pull/880, quote retrieval shouldn't be an issue on the backup side.

I'm happy to leave this as it is for now and we can fix the test and prioritise #1014 as you said when necessary.